ykcapi/src/lib.rs

```diff
@@ -25,6 +24,7 @@ use ykrt::{HotThreshold, Location, MT};
 use yksmp::{Location as SMLocation, StackMapParser};
 
 #[no_mangle]
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
```

Comment:
Is this related to: https://github.com/rust-lang/rust-clippy/issues/3045 ?

Reply:
Yes.

Reply:
Should we just let it warn? I worry we will forget to remove these annotations.

This would of course work against the idea of adding regular automated clippy checks...

Reply:
Notice this warning is only disabled for this one function.

Reply:
Yes, I know. Still...

Also we have many instances of this for other functions.

Reply:
The warning only triggers on this function. We definitely want to be warned of other uses, hence why I restricted to disabling it on this function alone.

Reply:
Yes, I understand. We are talking past each other maybe :)

If this function were to change such that the warning is no longer a false positive, but a true positive, then would it not be dangerous to have this annotation around?

Reply:
?

Reply:
The issue I linked explains that the clippy warning often triggers erroneously, as a false positive, right? As I understand, you are silencing a false positive here, right?

So what if a future code edit means that the clippy warning is actually justified, and that we've silenced a real concern?

Reply:
I'm asking: should we *not* suppress the false positive. We don't have to fix every warning.

Reply:
What's the point of a warning that we then ignore? It's literally worse than suppressing it!

Reply:
I give up. Please squash.